### ISSUE TITLE
[FEA] Add `.cut` api support

### ISSUE THREAD

**Issue by argenisleon** [rapidsai/cudf#5589]
**Is your feature request related to a problem? Please describe.**
I need to segment a series of continuous values. For pandas, I could use `.cut()`. I would like cudf to implement this.
```
pdf = pd.DataFrame({'numbers': [0,1,2,3,4,5,6,78,31,45,60,1]}, )
pd.cut(pdf.numbers, 2, include_lowest=True, labels=list(range(2)))
0     0
1     0
2     0
3     0
4     0
5     0
6     0
7     1
8     0
9     1
10    1
11    0
```

**Describe the solution you'd like**
Mirror pandas `.cut` behavior

**Describe alternatives you've considered**
`cp.digitize` seems to be a good alternative

**Comment by datametrician** [comment]
@kkraus14 given this is a recsys/customer ask could we get this on the roadmap for 0.17?

**Comment by kkraus14** [comment]
Depends on https://github.com/rapidsai/cudf/issues/5376

**Comment by kkraus14** [comment]
Depends on #7517

**Comment by beckernick** [comment]
Closing this issue as `cut` is now available in the 21.08 nightly:

```python
import pandas as pd
import cudf
​
df = cudf.DataFrame({'numbers': [0,1,2,3,4,5,6,78,31,45,60,1]}, )
res = cudf.cut(df.numbers, 2, include_lowest=True, labels=list(range(2)))
​
res.to_pandas().equals(
    pd.cut(pdf.numbers, 2, include_lowest=True, labels=list(range(2)))
)
True
```
